ui/helpers/utils/token-util.js

```diff
@@ -212,3 +215,55 @@ export function getTokenFiatAmount(
   }
   return result;
 }
+
+export async function getAssetDetails(
+  tokenAddress,
+  currentUserAddress,
+  transactionData,
+  existingCollectibles,
+) {
+  const tokenData = parseTransactionData(transactionData);
+  if (!tokenData) {
+    throw new Error('Unable to detect valid token data');
+  }
+
+  const tokenId = getTokenValueParam(tokenData);
+  let tokenDetails;
+  try {
+    tokenDetails = await getTokenStandardAndDetails(
+      tokenAddress,
+      currentUserAddress,
+      tokenId,
+    );
+  } catch (error) {
+    log.warn(error);
+    return {};
+  }
+  if (tokenDetails?.standard === ERC721 || tokenDetails?.standard === ERC1155) {
+    const existingCollectible = existingCollectibles.find(({ address }) =>
+      util.isEqualCaseInsensitive(tokenAddress, address),
+    );
+    if (existingCollectible) {
+      return {
+        address: existingCollectible?.address,
+        description: existingCollectible?.description,
+        favorite: existingCollectible?.favorite,
+        image: existingCollectible?.image,
+        isCurrentlyOwned: existingCollectible?.isCurrentlyOwned,
+        name: existingCollectible?.name,
+        tokenId: existingCollectible?.tokenId,
```

Comment:
Since you do an `existingCollectible` check, we don't need all of the `?.` property checks.  Also, since you're mirroring over many properties, can you simply return `{...existingCollectible, standard: tokenDetails?.standard}`?

Reply:
fixed here: https://github.com/MetaMask/metamask-extension/pull/13131/commits/8c1109875b2bd628c85dd5554e316bb2b53582a5